README.md
# uajs 多语言文档 zh-cn
#概述
uajs从UserAgent上获取关于如下的信息  
1.操作系统信息:可以获取操作系统和版本  
2.可以获取设备信息，仅在ios上，获取ipad/iphone/itouch  
3.应用信息：名称及版本  

#场景
1.判断用户是否使用手机，及使用的手机是否是android或者ios  
2.判断用户是否使用某些应用(微信 或者QQ)及版本

#支持的操作系统
1.android  
2.ios  
3.windows phone  
4.windows destop:windows的桌面版  
5.macosx  
6.xwindow

#支持的应用
1.wechat:微信  
2.weibo:微博客户端  
3.alipay:支付宝客户端  
4.qq:QQ客户端  
5.lbbrowser:猎豹浏览器  
6.firefox:火狐浏览器 (1.1+)  
7.ucbrowser:UC浏览器 (1.1+)  
8.mqqbrowser:QQ浏览器(移动端) (1.1+)  
9.baidubrowser:百度浏览器 (1.1+)  
10.mxbrowser:遨游云浏览器 (1.1+)   
11.liebaofast:猎豹浏览器(移动端) (1.1+)  
12.mb2345browser:2345浏览器(移动端) (1.1+)  
13.miuibrowser:小米手机浏览器 (1.1+)  
14.chrome:Chrome浏览器(1.1+)  
15.edge:Edge浏览器(1.1+)  
16.chitu:赤兔(1.2+)  
17.dingtalk:钉钉(1.2+)  
18.sogouexplorer:搜狗浏览器(1.2+)  
19.tbapp:淘宝app(1.2+)


#引用方式 && 版本说明
##调用
在head上引用:  
http://hstatic.cn/uajsapi/[ver/]uajsapi.js   
其中：  
如果ver/可选，如果不输入，则默认使用最新版本  
压缩版:
http://hstatic.cn/uajsapi/[ver/]uajsapi_compress.js   

##使用
<pre>
var uainfo=new UaInfo();   
document.writeln("操作系统:" + uainfo.osinfo.osname);    
document.writeln("操作系统版本:" + uainfo.osinfo.osver);   
document.writeln("设备名称:" + uainfo.deviceinfo);  
document.writeln("应用名称" + uainfo.appinfo.appname);   
document.writeln("应用版本:" + uainfo.appinfo.appver);   
if(uainfo.hasApplePay){
    document.writeln("support apple pay");
}

</pre>
测试链接：http://hstatic.cn/demo.html  
QR：![demo url](/demo_url.png)

#文档对象
uainfo包含如下属性  
1.osinfo :操作系统信息,包含如下属性  
osname:操作系统名称，见 支持的操作系统  
osver:操作系统的版本

2.deviceinfo:可能的设备名称，仅供参考  

3.appinfo:应用信息,包含如下属性  
appname:应用名称，见 支持的应用  
appver:应用版本

4.hasApplePay:是否支持apple Pay

#收集cookie
应用会通过cookie方式收集如下信息：  
1.用户的useragent串  
2.refer/ip  
3.每个用户分配一个uid


#联系方式
欢迎大家就加入，以便在变化时及时通知  
QQ群：QR：![demo url](/qqqun.png)  
群号：17924450 数据微支持群



#协议
BSD

changelog.md
#2016.3.8 1.1
1.处理当uainfo的osinfo appinfo默认为undefined的情况  
2.增加app识别库

#2016.3.14 1.2
1.osregxprule支持优先级,并调整部分规则  
2.补充 赤兔,钉钉,搜狗浏览器,淘宝app 

uajsapi/v1.2/uajsapi_compress.js
setUid();var UaInfo=function(){this.uastr=navigator.userAgent;this.uascreen=getscreeninfo();collectUaInfo(this);return this},UaInfo=function(a){this.uastr=a;this.uascreen=getscreeninfo();collectUaInfo(this);return this},UaInfo=function(a,b){this.uastr=a;this.uascreen=b;collectUaInfo(this);return this};logit(navigator.userAgent);
function collectUaInfo(a){"undefined"===typeof a.uastr&&(a.uastr=navigator.userAgent);"undefined"===typeof a.uascreen&&(a.uascreen=getscreeninfo());setDefaultUainfo(a);var b=new OSRegxpRule("android",["Android (.*?);"],[";.*;( .*) Build/"],1),c=new OSRegxpRule("ios",["OS (.*) like Mac OS X"],["\\((.*?);"],1),d=new OSRegxpRule("windows phone",["Windows Phone (.*?);"],[".*; (.*?)\\)"],10),e=new OSRegxpRule("windows destop",["Windows NT (.*?); "],[""],1),f=new OSRegxpRule("macosx",["Intel Mac OS X (.*?)\\)"],
[""],1),g=new OSRegxpRule("xwindow",[" \\((X11); Linux "],[""],1),e=[b,c,d,e,f,g];e.sort(function(a,b){return b.prio-a.prio});for(b=0;b<e.length;b++)if(c=e[b],d=getRegXpValue(a.uastr,c.osverregxps,1),0<d.length){b={};b.osname=c.osname;b.osver=d;a.osinfo=b;a.deviceinfo=getRegXpValue(a.uastr,c.deviceinforegxp,1);a.deviceinfo=a.deviceinfo.trim();a.deviceinfo=a.deviceinfo.toLowerCase();break}if(""!==a.osinfo){if("ios"===a.osinfo.osname||"macosx"===a.osinfo.osname)a.osinfo.osver=a.osinfo.osver.replace(/_/g,
".");c=[new AppRegxp("wechat","\u5fae\u4fe1",["MicroMessenger/(.*?) "],10),new AppRegxp("weibo","\u5fae\u535a\u624b\u673a\u7aef",["weibo__(.*?)__"],1),new AppRegxp("alipay","\u652f\u4ed8\u5b9d\u624b\u673a\u7aef",["AlipayClient/(.*?) "],2),new AppRegxp("qq","qq\u624b\u673a\u5ba2\u6237\u7aef",["QQ/(.*?) "],3),new AppRegxp("lbbrowser","\u730e\u8c79\u6d4f\u89c8\u5668",[" (LBBROWSER)"],9),new AppRegxp("firefox","\u706b\u72d0\u6d4f\u89c8\u5668",[" Firefox/(\\d+.\\d+) ","Firefox/(.*);","Firefox/(.*)"],1),
new AppRegxp("ucbrowser","UC\u6d4f\u89c8\u5668",["UCBrowser/(.*?) "],1),new AppRegxp("mqqbrowser","QQ\u6d4f\u89c8\u5668(\u79fb\u52a8\u7aef)",["MQQBrowser/(.*?) "],2),new AppRegxp("baidubrowser","\u767e\u5ea6\u6d4f\u89c8\u5668",["baidubrowser/(.*?) "],2),new AppRegxp("mxbrowser","\u9068\u6e38\u4e91\u6d4f\u89c8\u5668",["MxBrowser/(.*)"],1),new AppRegxp("liebaofast","\u730e\u8c79\u6d4f\u89c8\u5668(\u79fb\u52a8\u7aef)",["LieBaoFast/(.*)"],1),new AppRegxp("mb2345browser","2345\u6d4f\u89c8\u5668(\u79fb\u52a8\u7aef)",
["Mb2345Browser/(.*?) "],1),new AppRegxp("miuibrowser","\u5c0f\u7c73\u624b\u673a\u6d4f\u89c8\u5668",["MiuiBrowser/(.*)"],2),new AppRegxp("chrome","chrome",["Chrome/(.*?) "],1),new AppRegxp("edge","Edge",["Edge/(.*)"],2),new AppRegxp("chitu","\u8d64\u5154",["Chitu/(.*?) "],2),new AppRegxp("dingtalk","\u9489\u9489",["AliApp\\(DingTalk/(.*?)\\)"],2),new AppRegxp("sogouexplorer","\u641c\u72d7\u6d4f\u89c8\u5668",["SE (.*) MetaSr"],2),new AppRegxp("tbapp","\u6dd8\u5b9dapp",["AliApp\\(TB/(.*)\\)"],2)];c.sort(function(a,
b){return b.prio-a.prio});for(b=0;b<c.length;b++)if(d=c[b],e=getRegXpValue(a.uastr,d.appregxps,1),0<e.length){appinfo={};appinfo.appname=d.appname;appinfo.appver=e;a.appinfo=appinfo;break}"ios"===a.osinfo.osname&&("wechat"===a.appinfo.appname&&doitatwechat(a),"weibo"===a.appinfo.appname&&doitatweibo(a));setApplePay(a);logit(a.uastr)}}
function setApplePay(a){if("ios"===a.osinfo.osname){var b=["iPhone 6S Plus","iphone 6S","iphone 6 plus","iphone 6"],c=[];a.possdevice instanceof Array&&(c=a.possdevice);c.push(a.deviceinfo);for(var d=0;d<c.length;d++)if(in_array(b,c[d])){a.hasApplePay=!0;break}}}function setDefaultUainfo(a){a.osinfo="";a.deviceinfo="";a.osinfo={osname:"",osver:""};appinfo={appname:"",appver:""};a.appinfo=appinfo;a.hasApplePay=!1}function in_array(a,b){for(var c=0;c<a.length;c++)if(a[c]===b)return!0;return!1}
function doitatweibo(a){var b=getRegXpValue(a.uastr,"Weibo \\((.*)__weibo",1);if(0!==b.length){weidevicelist="iPhone8,2;iPhone 6S Plus;iPhone8,1;iphone 6S;iPhone7,1;iphone 6 plus;iPhone7,2;iphone 6;iPhone6,1;iphone 5s;iPhone6,2;iphone 5s;iPhone5,3;iphone 5c;iPhone5,4;iphone 5c;iPhone5,2;iphone 5;iPhone5,1;iphone 5;iPhone4,1;iphone 4s;iPhone3,3;iphone 4;iPhone3,2;iphone 4;iPhone3,1;iphone 4;iPad5,2;iPad mini 4;iPad5,1;iPad mini 4;iPad4,9;iPad mini 3;iPad4,8;iPad mini 3;iPad4,7;iPad mini 3;iPad4,6;iPad mini 2;iPad4,5;iPad mini 2;iPad4,4;iPad mini 2;iPad2,7;iPad mini;iPad2,6;iPad mini;iPad2,5;iPad mini;iPad5,4;iPad Air 2;iPad5,3;iPad Air 2;iPad4,3;iPad Air;iPad4,2;iPad Air;iPad4,1;iPad Air;iPad3,6;ipad 4;iPad3,5;ipad 4;iPad3,4;ipad 4;iPad3,3;the New iPad;iPad3,2;the New iPad;iPad3,1;the New iPad;iPad2,4;ipad 2;iPad2,3;ipad 2;iPad2,2;ipad 2;iPad2,1;ipad 2;iPad6,8;iPad Pro;iPad6,7;iPad Pro;iPad1,1;ipad 1".split(";");
for(var c=0;c<weidevicelist.length-1;c+=2)weidevicelist[c]===b&&(a.deviceinfo=weidevicelist[c+1])}}
function doitatwechat(a){if("iphone"===a.deviceinfo.toLowerCase())screeninfo="iphone 4;480,320;iphone 4s;480,320;iphone 5;568,320;iphone 5c;568,320;iphone 5s;568,320;iphone 6;667,375;iphone 6s;667,375;iphone 6 plus;736,414;iphone 6 plus;736,414".split(";");else if("ipad"===a.deviceinfo.toLowerCase())screeninfo="ipad 1;1024,768;ipad 2;1024,768;the New iPad;2048,1536;ipad 4;2048,1536;iPad Air;2048,1536;iPad mini;1024,768;iPad Air 2;2048,1536;iPad mini 2;2048,1536;iPad mini 3;2048,1536;iPad mini 4;2048,1536;iPad Pro;2732,2048".split(";");else return;
var b=[];basicscreen=a.uascreen;tagscreen=basicscreen[0]+","+basicscreen[1];for(var c=1;c<screeninfo.length-1;c+=2)screeninfo[c]===tagscreen&&b.push(screeninfo[c-1]);1<b.length&&(a.possdevice=b)}function AppRegxp(a,b,c,d){this.appname=a;this.appdesc=b;this.appregxps=c;this.prio=d;return this}function OSRegxpRule(a,b,c,d){this.osname=a;this.osverregxps=b;this.deviceinforegxp=c;this.prio=d;return this}
function logit(a){"localhost"!==location.hostname&&(screeninfo=getscreeninfo(),uid=getCookie("uid"),picurl="http://col.datatiny.com/log/?screeninfo="+screeninfo+"&uid="+uid,a!==navigator.userAgent&&(picurl=picurl+"&uastr="+encodeURIComponent(a)),crPic(picurl))}function setUid(){if(!(0<getCookie("uuid").length)){var a=document.createElement("script");a.type="text/javascript";a.src="http://cruid.datatiny.com/uid/";document.getElementsByTagName("head")[0].appendChild(a)}}
function crPic(a){var b=document.getElementById("datatinypic");null===b&&(b=document.createElement("img"),b.id="datatinypic",b.style.display="none",document.getElementsByTagName("head")[0].appendChild(b));b.src=a}
function getscreeninfo(){var a=[];a.push(screen.height);a.push(screen.width);a.push(screen.availWidth);a.push(screen.availHeight);a.push(screen.colorDepth);a.push(screen.bufferDepth);a.push(screen.deviceXDPI);a.push(screen.deviceYDPI);a.push(screen.logicalXDPI);a.push(screen.logicalYDPI);a.push(screen.fontSmoothingEnabled);a.push(screen.pixelDepth);a.push(screen.updateInterval);return a}
function setcookieY(a,b,c,d){c=new Date;c.setFullYear(c.getFullYear()+50);cookiestr=a+"="+b+";expires="+c+";domain="+d;document.cookie=cookiestr}function getCookie(a){return 0<document.cookie.length&&(c_start=document.cookie.indexOf(a+"="),-1!==c_start)?(c_start=c_start+a.length+1,c_end=document.cookie.indexOf(";",c_start),-1===c_end&&(c_end=document.cookie.length),unescape(document.cookie.substring(c_start,c_end))):""}
function getRegXpValue(a,b,c){c=[];c=b instanceof Array?b:Array(b);if(0===c.length)return"";for(b=0;b<c.length;b++)if(arr=(new RegExp(c[b],"gi")).exec(a),arr instanceof Array&&1<arr.length)return arr[1];return""};

uajsapi/v1.2/uajsapi.js
/*
 *ver:1.2 doc:
 */
 setUid();
var UaInfo=function(){
    this.uastr=navigator.userAgent;
    this.uascreen=getscreeninfo();
    collectUaInfo(this);
    return this;
}
var UaInfo=function(uastr){
    this.uastr=uastr;

    this.uascreen=getscreeninfo();
     collectUaInfo(this);
    return this;
}
var UaInfo=function(uastr,uascreen){
    this.uastr=uastr;
    this.uascreen=uascreen;
     collectUaInfo(this);
    return this;
    
}


 logit(navigator.userAgent);
function collectUaInfo(uainfo){

    if(typeof uainfo.uastr === 'undefined') {
        uainfo.uastr=navigator.userAgent;
    }
    if(typeof uainfo.uascreen === 'undefined') {
        uainfo.uascreen=getscreeninfo();		
    }
    //default uainfo
    setDefaultUainfo(uainfo);
    var android=new OSRegxpRule("android",new Array("Android (.*?);"),new Array(";.*;( .*) Build\/"),1);
    var ios=new OSRegxpRule("ios",new Array("OS (.*) like Mac OS X"),new Array("\\((.*?);"),1); 
    var wp=new OSRegxpRule("windows phone",new Array("Windows Phone (.*?);"),new Array(".*; (.*?)\\)"),10);
    var windows_destop=new OSRegxpRule("windows destop",new Array("Windows NT (.*?); "),new Array(""),1);
    var macos=new OSRegxpRule("macosx",new Array("Intel Mac OS X (.*?)\\)"),new Array(""),1);
    var xwindow=new OSRegxpRule("xwindow",new Array(" \\((X11); Linux "),new Array(""),1);
    var osregxps=new Array(android,ios,wp,windows_destop,macos,xwindow);
   
     osregxps.sort(function(a,b){
         return b.prio-a.prio;
    });
    
    for(var i=0;i<osregxps.length;i++){
       var osregxp=osregxps[i];
     
        var osver=getRegXpValue(uainfo.uastr,osregxp.osverregxps,1);
     
        if(osver.length>0){
            var osinfo=new Object();
            osinfo.osname=osregxp.osname;
            osinfo.osver=osver;
            uainfo.osinfo=osinfo;
            uainfo.deviceinfo=getRegXpValue(uainfo.uastr,osregxp.deviceinforegxp,1);
            uainfo.deviceinfo=uainfo.deviceinfo.trim();
            uainfo.deviceinfo=uainfo.deviceinfo.toLowerCase();
            break;
        }
    }

    //if not found osinfo in UA then return ;
    if(uainfo.osinfo === '') return ;
    
    if(uainfo.osinfo.osname==="ios" || uainfo.osinfo.osname==="macosx" ){
        uainfo.osinfo.osver=uainfo.osinfo.osver.replace(/_/g,".");
    }
     
     
    //app regxp支持优先级,越高越优先
    var appregxps=new Array(new AppRegxp("wechat","微信",new Array("MicroMessenger/(.*?) "),10),
        new AppRegxp("weibo","微博手机端",new Array("weibo__(.*?)__"),1),
        new AppRegxp("alipay","支付宝手机端",new Array("AlipayClient/(.*?) "),2),
        new AppRegxp("qq","qq手机客户端",new Array("QQ/(.*?) "),3),
        new AppRegxp("lbbrowser","猎豹浏览器",new Array(" (LBBROWSER)"),9),
        new AppRegxp("firefox","火狐浏览器",new Array(" Firefox/(\\d+\.\\d+) ","Firefox/(.*);","Firefox/(.*)"),1),
        new AppRegxp("ucbrowser","UC浏览器",new Array("UCBrowser/(.*?) "),1),
        new AppRegxp("mqqbrowser","QQ浏览器(移动端)",new Array("MQQBrowser/(.*?) "),2),
        new AppRegxp("baidubrowser","百度浏览器",new Array("baidubrowser/(.*?) "),2),
        new AppRegxp("mxbrowser","遨游云浏览器",new Array("MxBrowser/(.*)"),1),
        new AppRegxp("liebaofast","猎豹浏览器(移动端)",new Array("LieBaoFast/(.*)"),1),
        new AppRegxp("mb2345browser","2345浏览器(移动端)",new Array("Mb2345Browser/(.*?) "),1),
        new AppRegxp("miuibrowser","小米手机浏览器",new Array("MiuiBrowser/(.*)"),2),
        new AppRegxp("chrome","chrome",new Array("Chrome/(.*?) "),1),
        new AppRegxp("edge","Edge",new Array("Edge/(.*)"),2),
        new AppRegxp("chitu","赤兔",new Array("Chitu/(.*?) "),2),
        new AppRegxp("dingtalk","钉钉",new Array("AliApp\\(DingTalk/(.*?)\\)"),2),
        new AppRegxp("sogouexplorer","搜狗浏览器",new Array("SE (.*) MetaSr"),2),
        new AppRegxp("tbapp","淘宝app",new Array("AliApp\\(TB/(.*)\\)"),2)
        
        );
     //sort by prio
     appregxps.sort(function(a,b){
         return b.prio-a.prio;
    });
    //get app info
    for(var i=0;i<appregxps.length;i++){
        var appregxp=appregxps[i];
        var appversion=getRegXpValue(uainfo.uastr,appregxp.appregxps,1);
        if(appversion.length >0){
            appinfo=new Object();
            appinfo.appname=appregxp.appname;
            appinfo.appver=appversion;
            uainfo.appinfo=appinfo;
            break;
        }
    }

    //if os is ios ,device is apple ,
    //wechat or weibo
    if(uainfo.osinfo.osname ==="ios"){
       
        if(uainfo.appinfo.appname==="wechat"){doitatwechat(uainfo);}
        if(uainfo.appinfo.appname==="weibo"){doitatweibo(uainfo);}
    }

    //has apple pay
    setApplePay(uainfo);
    logit(uainfo.uastr);
}



function setApplePay(uainfo){
    if(uainfo.osinfo.osname!=="ios") return ;
    var applepaydevicelist=new Array("iPhone 6S Plus","iphone 6S","iphone 6 plus","iphone 6");
    var ar=new Array();
    if(uainfo.possdevice instanceof Array)ar=uainfo.possdevice;
    ar.push(uainfo.deviceinfo);
    for(var i =0;i<ar.length;i++){
        if(in_array(applepaydevicelist,ar[i])){
            uainfo.hasApplePay=true;
            return ;
        }
    }
   
}

function setDefaultUainfo(uainfo){
var osinfo=new Object();
osinfo.osname="";
osinfo.osver="";
uainfo.osinfo="";
uainfo.deviceinfo="";
uainfo.osinfo=osinfo;
appinfo=new Object();
appinfo.appname="";
appinfo.appver="";
uainfo.appinfo=appinfo;
uainfo.hasApplePay=false;  
}
function in_array(arr,sub){
    for(var i =0;i<arr.length;i++){
        if(arr[i]===sub) return true ;
    }
    return false;
}

function doitatweibo(uainfo){
    var weibodevice=getRegXpValue(uainfo.uastr,"Weibo \\((.*)__weibo",1);

    if(weibodevice.length===0) return ;
    weidevicelist=new Array("iPhone8,2","iPhone 6S Plus","iPhone8,1","iphone 6S","iPhone7,1","iphone 6 plus",
    "iPhone7,2","iphone 6","iPhone6,1","iphone 5s","iPhone6,2","iphone 5s","iPhone5,3","iphone 5c",
    "iPhone5,4","iphone 5c","iPhone5,2","iphone 5","iPhone5,1","iphone 5","iPhone4,1","iphone 4s",
    "iPhone3,3","iphone 4","iPhone3,2","iphone 4","iPhone3,1","iphone 4",
    "iPad5,2","iPad mini 4", "iPad5,1","iPad mini 4","iPad4,9","iPad mini 3","iPad4,8","iPad mini 3",
    "iPad4,7","iPad mini 3", "iPad4,6","iPad mini 2","iPad4,5","iPad mini 2","iPad4,4","iPad mini 2",
    "iPad2,7","iPad mini", "iPad2,6","iPad mini", "iPad2,5","iPad mini","iPad5,4","iPad Air 2",
    "iPad5,3","iPad Air 2","iPad4,3","iPad Air","iPad4,2","iPad Air","iPad4,1","iPad Air",
    "iPad3,6","ipad 4","iPad3,5","ipad 4","iPad3,4","ipad 4","iPad3,3","the New iPad",
    "iPad3,2","the New iPad","iPad3,1","the New iPad","iPad2,4","ipad 2","iPad2,3","ipad 2",
    "iPad2,2","ipad 2","iPad2,1","ipad 2","iPad6,8","iPad Pro","iPad6,7","iPad Pro",
    "iPad1,1","ipad 1");
    for(var i=0;i<weidevicelist.length-1;i=i+2){
        if(weidevicelist[i]===weibodevice) {
            uainfo.deviceinfo= weidevicelist[i+1];
          }
        
    }
}

//wechat
function doitatwechat(uainfo){

    if(uainfo.deviceinfo.toLowerCase()==="iphone"){
        screeninfo=new Array("iphone 4","480,320","iphone 4s","480,320",
            "iphone 5","568,320","iphone 5c","568,320","iphone 5s","568,320",
            "iphone 6","667,375","iphone 6s","667,375",
            "iphone 6 plus","736,414","iphone 6 plus","736,414");
    }else if(uainfo.deviceinfo.toLowerCase()==="ipad"){
        screeninfo=new Array("ipad 1","1024,768","ipad 2","1024,768",
            "the New iPad","2048,1536","ipad 4","2048,1536",
            "iPad Air","2048,1536","iPad mini","1024,768",
            "iPad Air 2","2048,1536","iPad mini 2","2048,1536",
            "iPad mini 3","2048,1536","iPad mini 4","2048,1536",
            "iPad Pro","2732,2048");
    }else{
        return ;
    }

    var possdevice=new Array();
    basicscreen=uainfo.uascreen;
    tagscreen=basicscreen[0]+","+basicscreen[1];

    for(var i=1;i<screeninfo.length-1;i+=2){
        if(screeninfo[i]===tagscreen){
          
           possdevice.push(screeninfo[i-1]);
        }
    }
    if(possdevice.length>1){
        uainfo.possdevice=possdevice;
    }
     
}

function AppRegxp(appname,appdesc,appregxps,prio){
    this.appname=appname;
    this.appdesc=appdesc;
    this.appregxps=appregxps;
    this.prio=prio;
    return this;
    
}
function OSRegxpRule(osname,osverregxp,deviceinforegxp,prio){
    this.osname=osname;
    this.osverregxps=osverregxp;
    this.deviceinforegxp=deviceinforegxp;
    this.prio=prio;
    return this;
}

function logit(uastr){
    if(location.hostname==="localhost") return ;
  
    screeninfo=getscreeninfo();
    uid=getCookie("uid");
    picurl="http://col.datatiny.com/log/?screeninfo="+screeninfo+"&uid="+uid ;
    if(uastr !== navigator.userAgent){
        picurl=picurl + "&uastr="+encodeURIComponent(uastr) ;
    }
    crPic(picurl);
}

function setUid(){
     if(getCookie("uuid").length>0){
        return ;
    }
    var url="http://cruid.datatiny.com/uid/";
    var ele=document.createElement("script");
    ele.type="text/javascript";
    ele.src=url;
    document.getElementsByTagName("head")[0].appendChild(ele);
}
function crPic(picurl){
   var picele=document.getElementById("datatinypic");
   if(picele ===null){
     var ele=document.createElement("img");
     ele.id="datatinypic";
     ele.style.display="none";
     document.getElementsByTagName("head")[0].appendChild(ele);
     picele=ele;
   }
    picele.src=picurl;
  
}

function getscreeninfo(){
    var screeninfo=new Array();
    screeninfo.push(screen.height);
    screeninfo.push(screen.width);
    screeninfo.push(screen.availWidth);
    screeninfo.push(screen.availHeight);
    screeninfo.push(screen.colorDepth);
    screeninfo.push(screen.bufferDepth);
    screeninfo.push(screen.deviceXDPI);
    screeninfo.push(screen.deviceYDPI);
    screeninfo.push(screen.logicalXDPI);
    screeninfo.push(screen.logicalYDPI);
    screeninfo.push(screen.fontSmoothingEnabled);
    screeninfo.push(screen.pixelDepth);
    screeninfo.push(screen.updateInterval);

    return screeninfo;
}
function setcookieY(cookiename,cookvaile,expire,domain){
    var d=new Date();
    //defualt 50years ~
    d.setFullYear(d.getFullYear()+50);
    cookiestr=cookiename+"=" + cookvaile+";expires=" + d + ";domain=" +domain;
    document.cookie=cookiestr;
}
function getCookie(c_name){
  if (document.cookie.length>0){
        c_start=document.cookie.indexOf(c_name + "=");
    if (c_start!==-1){ 
        c_start=c_start + c_name.length+1;
        c_end=document.cookie.indexOf(";",c_start);
    if (c_end===-1) c_end=document.cookie.length;
    return unescape(document.cookie.substring(c_start,c_end));
  } 
  }
  return "";
}

  function getRegXpValue(str,regxpvalue,index){
      
        var dataregxp=new Array();
          if(regxpvalue instanceof Array){
             dataregxp= regxpvalue;
          }else{
              dataregxp=new Array(regxpvalue);
          }
          if(dataregxp.length===0) return "";
          for(var i=0;i<dataregxp.length;i++){
              var patt = new RegExp(dataregxp[i],"gi");
              arr = patt.exec(str); 
              if(arr instanceof Array &&  arr.length>1){
                 
                  return (arr[1]);
              }
          }
         return "";
}